packages/frontend/src/pages/VocabularyPracticePage.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import { get } from 'aws-amplify/api';
+import { toJSON } from '../utilities';
+
+// Define types for the response to ensure type safety
+interface Question {
+  question: string;
+  choices: string[];
+  right_choice: string;
+  explanation: string;
+}
+
+const VocabularyPracticePage: React.FC = () => {
+  const [questionType, setQuestionType] = useState<string>('vocabA1');
+  const [question, setQuestion] = useState<Question | null>(null);
+  const [selectedAnswer, setSelectedAnswer] = useState<string>('');
+  const [showExplanation, setShowExplanation] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+  const [initialFetch, setInitialFetch] = useState<boolean>(true);
+
+  const fetchQuestion = async () => {
+    try {
+      const response = await toJSON(
+        get({
+          apiName: 'myAPI',
+          path: `/question/${questionType}`,
+        }),
+      );
+      setQuestion(response);
+      setSelectedAnswer(''); // Reset selected answer on new question fetch
+      setShowExplanation(false); // Reset explanation visibility
+      setInitialFetch(false); // Remove initial fetch status
+    } catch (error: any) {
+      console.error('Error fetching question:', error);
+      setError(error.message || 'Unknown error');
+    }
+  };
+
+  const handleAnswer = (choice: string) => {
+    setSelectedAnswer(choice);
+    // Show explanation only if the chosen answer is not the right one
+    setShowExplanation(choice !== question?.right_choice);
```

Comment:
I think this is unnecessary, we should show the explanation regardless of whether the response is right or wrong.